src/pages/cw-storage-plus/containers/indexed-map.mdx

```diff
@@ -0,0 +1,244 @@
+import { Callout } from "nextra/components";
+
+# `IndexedMap`
+
+An
+[`IndexedMap`](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.IndexedMap.html)
+is a map that, apart from the usual map key (`K`) has some secondary indexes
+(`I`) that can be used to look up values.
+
+<Callout>
+There's no limit to how many indexes you can have, but be careful. Using many indexes can increase the complexity of
+storage writes - with every write, the list of indexes is iterated over since
+they might need to be updated.
+</Callout>
+
+As always, we encourage you to explore the
+[`API reference`](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.IndexedMap.html)
+for a full list of available methods and more rigid definitions of the types
+involved.
+
+## Case study: User lookup
+
+Imagine we have a list of users. Every user is uniquely identified by their
+address, so it makes sense to use that as the primary key. We'll need to look up
+users by their address - if Alice calls the contract, we should be able to look
+up her user data by the address with which she called.
+
+<Callout>
+Every snippet on this page builds on the previous ones. Make sure you've
+  understood the previous snippet before moving on to the next one, and feel free to go back and forth as needed.
+</Callout>
+
+This is easy to model with a normal `Map`.
+
+```rust template="storage"
+use cw_storage_plus::Map;
+
+type Handle = String;
+
+#[cw_serde]
+struct User {
+    handle: Handle,
+    country: String,
+}
+
+let _users = Map::<Addr, User>::new("u");
+```
+
+Great! But what if we want to look up users by their handles? Our only real
+option here is to iterate over all users and check if the handle matches. With a
+big enough user base, this could be slow and expensive.
+
+### Setting up an `IndexedMap` with a `UniqueIndex`
+
+A better way would be to maintain a secondary index. Let's try to do this.
+
+```rust template="storage"
+use cw_storage_plus::{index_list, IndexedMap, UniqueIndex};
+
+use crate::users::*;
+
+#[index_list(User)]
+struct UserIndexes<'a> {
+    handle_ix: UniqueIndex<'a, Handle, User, Addr>,
```

Comment:
how do the type arguments here relate to the index? i.e. `Addr` was noted as being primary... how does that relate to the signature here, is primary always the last?

Reply:
yeah, I think it's worth noting